testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/assertions/ElementAssertj.java

```diff
@@ -229,6 +230,65 @@ public void hasActiveElementsExactly(
         });
   }
 
+  public void hasCompletedElementsInOrder(
+      final long processInstanceKey, final List<ElementSelector> elementSelectors) {
+
+    awaitFlowNodeInstanceAssertion(
+        processInstanceFilter(processInstanceKey),
+        flowNodeInstances -> {
+          final List<FlowNodeInstance> completedFlowNodeInstances =
+              getInstancesInState(flowNodeInstances, FlowNodeInstanceState.COMPLETED);
+
+          final List<FlowNodeInstance> relevantFlowNodeInstances =
+              completedFlowNodeInstances.stream()
+                  .filter(i -> elementSelectors.stream().anyMatch(e -> e.test(i)))
+                  .collect(Collectors.toList());
+
+          if (elementSelectors.size() > relevantFlowNodeInstances.size()) {
+            fail(
+                String.format(
+                    "%s should have completed elements %s in order, but only the following elements were completed:\n%s",
+                    actual,
+                    formatElementSelectors(elementSelectors),
+                    formatFlowNodeInstanceStates(
+                        relevantFlowNodeInstances, FlowNodeInstance::getFlowNodeId)));
+          }
```

Comment:
:wrench: This check is unnecessary because of the `IntStream.rangeClosed` below. We can remove it.